### PR TITLE
feat: Starlark processor example for processing sparkplug_b messages

### PR DESCRIPTION
**Pull request by mtnguru**
### Required for all PRs:

- [ ] Updated associated README.md.
- [X] Wrote appropriate unit tests.

This PR adds an example for using the starlark processor  to process sparkplug_b messages. When DBIRTH messages are received a list of metric names is created with a numeric alias. When subsequent DDATA messages are received the numeric alias is used to find the metric name. In addition this processor creates additional influxdb tags from the topic and metric name

### CODE REVIEW

**Comment by mtnguru** [comment]
This starlark processor processes sparkplug_b messages.  When DBIRTH messages are received a list of metric names is created with a numeric alias.  When subsequent DDATA messages are received the numeric alias is used to find the metric name.  In addition this processor creates additional influxdb tags from the topic and metric name.

File: plugins/processors/starlark/testdata/sparkplug.star [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+                    metric.tags["Device"] = sfields[2]
+                if nf > 4:
+                    metric.tags["Position"] = sfields[3]
+                if nf > 5:
+                    metric.tags["Composition"] = sfields[4]
+
+def apply(metric):
+    output = metric
+
+#   log.debug("apply metric: {}".format(metric))
```
Comment (srebhan) [inline plugins/processors/starlark/testdata/sparkplug.star:235]: Please either remove those lines or keep and uncomment them.

File: plugins/processors/starlark/testdata/sparkplug.star [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    "devices":    dict(),
+    "unresolved": list()
+}
+
+def extractTopicTags(metric):
+    msg_format   = 'xx'
+    groupid      = 'xx'
+    msg_type     = 'xx'
+    edgeid       = 'xx'
+    deviceid     = 'xx'
```
Comment (srebhan) [inline plugins/processors/starlark/testdata/sparkplug.star:128]: Is this a placeholder? If so, please use some more illustrative values.

File: plugins/processors/starlark/testdata/sparkplug.star [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    if nfields > 0:
+      msg_format = fields[0]
+      if nfields > 1:
+        groupid = fields[1]
+        if nfields > 2:
+          msg_type = fields[2]
+          if nfields > 3:
+            edgeid = fields[3]
+            if nfields > 4:
+              deviceid = fields[4]
```
Comment (srebhan) [inline plugins/processors/starlark/testdata/sparkplug.star:142]: Please save some indention lines
```suggestion
    if nfields > 0:
      msg_format = fields[0]
    if nfields > 1:
      groupid = fields[1]
    if nfields > 2:
      msg_type = fields[2]
    if nfields > 3:
      edgeid = fields[3]
    if nfields > 4:
      deviceid = fields[4]
```

File: plugins/processors/starlark/testdata/sparkplug.star [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+                    metric.name = "Name"
+                else:
+                    metric.name = sfields[nf-1]     # The Units are used for the metric name
+                    metric.tags["Component"] = sfields[1]
+                if nf > 3:
+                    metric.tags["Device"] = sfields[2]
+                if nf > 4:
+                    metric.tags["Position"] = sfields[3]
+                if nf > 5:
+                    metric.tags["Composition"] = sfields[4]
```
Comment (srebhan) [inline plugins/processors/starlark/testdata/sparkplug.star:230]: Please reduce indention by one level. I know it costs some more cycles (as we need to check 2 conditions) but the readability is (especially for that example) more important.

**Review by srebhan — COMMENTED** [review]
Hey @mtnguru, thanks for committing this cool example!
The code looks very good overall, but I have some minor comments in the code, please take a look. 

Furthermore, please copy the text of your second post (explaining the PR) into your PR description above. And please remove the `resolves #` line as it might confuse the bot when merging.

File: plugins/processors/starlark/testdata/sparkplug.star (hunk 2 above)
Comment (mtnguru) [reply plugins/processors/starlark/testdata/sparkplug.star:128]: I changed them to a null string.

**Comment by srebhan** [comment]
@mtnguru I changed the topic and description to adhere to the usual form. Hope you don't mind.

Once you changed your code to resolve my comments, please push the changes to your branch to update the PR and let me know you are ready for another round. ;-)

**Comment by mtnguru** [comment]
I've pushed your recommended changes and they're ready for review.

**Comment by srebhan** [comment]
@mtnguru hmmm I don't see them here. Did you push to

> Add more commits by pushing to the master branch on mtnguru/telegraf.

? You should see the commits in this PR then...

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @mtnguru for providing this nice example as it will help a lot of SparkPlugB users!

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**